cinn/hlir/op/transform.cc

```diff
@@ -597,30 +596,39 @@ std::shared_ptr<OpStrategy> StrategyForMul(const framework::NodeAttr &attrs,
       ir::ModuleExpr mod_expr(vec_ast);
       ir::IRSchedule ir_sch(mod_expr);
       ir_sch.MergeExprs();
-      if (target.arch == Target::Arch::NVGPU) {
-        pe::IRCudaScheduleMul(ir_sch, output_shape, target);
-      } else if (target.arch == Target::Arch::X86) {
-#ifndef CINN_WITH_MKL_CBLAS
-        pe::IRMulScheduleCPU(ir_sch, temp_reduce_shape, target);
-#endif
+      auto blocks = ir_sch.GetAllBlocks();
+
+      int prod_size = std::accumulate(output_shapes[0].begin(), output_shapes[0].end(), 1, std::multiplies<int>());
+      if (prod_size > 1) {
+        if (ir_sch.GetLoops(blocks[0]).size() == 1) {
+          ir_sch.Bind(ir_sch.GetLoops(blocks[0])[0], "threadIdx.x");
+        } else {
+          ir_sch.Bind(ir_sch.GetLoops(blocks[0])[0], "blockIdx.x");
+          ir_sch.Bind(ir_sch.GetLoops(blocks[0])[1], "threadIdx.x");
+        }
       }
-      std::vector<CINNValue> res{CINNValue(ir_sch.GetModule().GetExprs().at(0))};
-      *ret = CINNValuePack{res};
-    } else {
-      CHECK(!args.empty()) << "The input argument of mul schedule is empty! Please check.\n";
-      CINNValuePack arg_pack = args[0];
-      Expr out               = arg_pack[0];
-      poly::StageMap stages  = arg_pack[1];
 
-      CHECK(out.as_tensor());
+      std::vector<CINNValue> results = {CINNValue(ir_sch.GetModule().GetExprs().at(0))};
+      *ret                           = CINNValuePack({results});
+    } else {
+      CHECK(arg_pack.size() == 2UL || arg_pack.size() == 3UL);
+      poly::StageMap stages = arg_pack.back();
       if (target.arch == Target::Arch::NVGPU) {
-        pe::CudaScheduleMul(stages, out.as_tensor_ref(), temp_reduce_shape, target);
```

Comment:
Do you inline the `CudaScheduleMul` code into here?

Do we still use `CudaScheduleMul`? If not, should we delete the`CudaScheduleMul` code?

Reply:
No, because `mul` is the same as `matmul` except the attribute different, so  here I just copy the `matmul`'s compute and schedule code into here.

I will pack the code into a common function later.